C2SM/processing-chain
# Refactoring (part 1)

### [Updating and consolidating Processing Chain for ICON-ART](https://github.com/C2SM/Tasks-Support/issues/12) (C2SM Task WG2 in 2023/1)

With the increasing number of supported model configurations and simulation options, parts of the processing chain (especially the main routine `run_chain.py`) have become rather complex and somewhat convoluted. 

The following parts of refactoring the code are addressed in this PR:

- [x] change naming: `target` -> `model`, `subtarget` -> `variant`
- [x] provide input files as dictionary (later to be replaced by `config.yaml` file)
- [x] introduce `models.yaml` file for global model configurations
- [x] reduce size of input data for test cases
- [x] job scripts: target models directly by their name
- [x] `run_chain.py`: distinguish features and variants instead of models
- [x] Remove non-test cases
- [x] add example test case for icon-art
- [x] add example test case for icon-art-global


**This is part 1 of the refactoring task.** Part 2 will mainly consist of the following:

- Change case configuration files to a more "config-like" format: `config.py` -> `config.yaml`
- Docs: Revision, updating, publish to github.io
- Add icon-art-oem-ensemble test case
- Check if `variants` and `features` can be unified
- Further quality-of-life changes for more intuitive handling of the chain
